### PR TITLE
Added new Trustar Integration

### PR DESCRIPTION
**Pull request by rod-castillo**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [X] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
N/A

## Description
Added a new integration to the TruStar pack to send data from TruStar to Intsights

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [X] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
   - [X] No

## Must have
- [X] Tests
- [X] Documentation

### CODE REVIEW

File: Packs/TruSTAR/Integrations/dw_send_trustar_data_to_intsights_copy/dw_send_trustar_data_to_intsights_copy.py [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+
+        else:
+            sendError('Unable to add IOCs document - {}'.format(response.content))
+    except Exception as e:
+        sendError('Unable to add IOCs document - {}'.format(e))
+
+
+# The command demisto.command() holds the command sent from the user.
+if demisto.command() == 'trustar-get-IOC-data':
+    uncleanListOfIndicators = get_ioc_data()
```
Comment (abaumgarten) [inline Packs/TruSTAR/Integrations/dw_send_trustar_data_to_intsights_copy/dw_send_trustar_data_to_intsights_copy.py:368]: Please note that `get_ioc_data ` might fail . As commands must raise exceptions on failure, you should catch those exceptions and use `return_error ` in order to handle it correctly.

File: Packs/TruSTAR/Integrations/dw_send_trustar_data_to_intsights_copy/dw_send_trustar_data_to_intsights_copy.py [hunk 2]
```diff
@@ -0,0 +1,406 @@
...
+    lengthOfUncleanList = len(uncleanListOfIndicators)
+    demisto.results(lengthOfUncleanList)
+    demisto.results("The IOCs that were Obtained...")
+    demisto.results({'ContentsFormat': formats['table'], 'Type': entryTypes['note'], 'Contents': uncleanListOfIndicators})
```
Comment (abaumgarten) [inline Packs/TruSTAR/Integrations/dw_send_trustar_data_to_intsights_copy/dw_send_trustar_data_to_intsights_copy.py:372]: This is not a good practice of returning results from a command. Please read about `CommandResults` here: https://xsoar.pan.dev/docs/integrations/code-conventions#commandresults

File: Packs/TruSTAR/Integrations/dw_send_trustar_data_to_intsights_copy/dw_send_trustar_data_to_intsights_copy.py [hunk 3]
```diff
@@ -0,0 +1,406 @@
...
+        indicator = i  # .to_dict(True)
+        itype = indicator['indicatorType'].lower()
+        demisto.info("Indicator: " + str(indicator) + " - indicator_type (lowercase): " + itype)
+
+        # Chcks to see if indicator already exists
+        if i not in cleanListOfIndicators:
+
+            # Make calls to the check functions for URLs, emails, IPv4 addresses, IPv6 addresses
+            if itype == 'ipv4' or itype == 'ipv6' or itype == 'ip':
+                if validateIP(indicator['value']) is True:
```
Comment (abaumgarten) [inline Packs/TruSTAR/Integrations/dw_send_trustar_data_to_intsights_copy/dw_send_trustar_data_to_intsights_copy.py:169]: What will happen here (as well as in other places in this function) if  `value` does not exist in `indicator`?
My suggestion is to use `get` (for avoiding an unexpected exception):
```suggestion
                if validateIP(indicator.get('value')) is True:
```

File: Packs/TruSTAR/Integrations/dw_send_trustar_data_to_intsights_copy/dw_send_trustar_data_to_intsights_copy.py [hunk 4]
```diff
@@ -0,0 +1,406 @@
...
+    demisto.results("The IOCs that were Obtained...")
+    demisto.results({'ContentsFormat': formats['table'], 'Type': entryTypes['note'], 'Contents': uncleanListOfIndicators})
+    demisto.setIntegrationContext(uncleanListOfIndicators)
+
+if demisto.command() == 'trustar-clean-IOC-data':
+    getUncleanList = demisto.getIntegrationContext()
+    cleanListOfIndicators = cleanData(getUncleanList)
+    demisto.results({'ContentsFormat': formats['table'], 'Type': entryTypes['note'], 'Contents': cleanListOfIndicators})
+    lengthOfCleanList = len(cleanListOfIndicators)
+    demisto.results(lengthOfCleanList)
```
Comment (abaumgarten) [inline Packs/TruSTAR/Integrations/dw_send_trustar_data_to_intsights_copy/dw_send_trustar_data_to_intsights_copy.py:380]: Please see my comment about using `CommandResults`

File: Packs/TruSTAR/Integrations/dw_send_trustar_data_to_intsights_copy/dw_send_trustar_data_to_intsights_copy.py [hunk 5]
```diff
@@ -0,0 +1,406 @@
...
+    # sendData(cleanListOfIndicators, doc_source_name)
+
+
+def sendData(data, title):
+    demisto.info("Sending Data: " + title)
+    demisto.results(len(data))
+    iocs = []
+    for item in data:
+        itype = item['indicatorType'].lower()
+        indicator = item['value']
```
Comment (abaumgarten) [inline Packs/TruSTAR/Integrations/dw_send_trustar_data_to_intsights_copy/dw_send_trustar_data_to_intsights_copy.py:276]:
```suggestion
        itype = item.get('indicatorType').lower()
        indicator = item.get('value')
```

File: Packs/TruSTAR/Integrations/dw_send_trustar_data_to_intsights_copy/dw_send_trustar_data_to_intsights_copy.py [hunk 6]
```diff
@@ -0,0 +1,406 @@
...
+    demisto.results({'ContentsFormat': formats['table'], 'Type': entryTypes['note'], 'Contents': cleanListOfIndicators})
+    lengthOfCleanList = len(cleanListOfIndicators)
+    demisto.results(lengthOfCleanList)
+    demisto.setIntegrationContext(cleanListOfIndicators)
+
+if demisto.command() == 'trustar-send-IOC-data-to-intsights':
+    getCleanList = demisto.getIntegrationContext()
+    demisto.results("Running TruSTAR IOC Data to Intsights...")
+    demisto.results(len(getCleanList))
+    sendData(getCleanList, doc_source_name)
```
Comment (abaumgarten) [inline Packs/TruSTAR/Integrations/dw_send_trustar_data_to_intsights_copy/dw_send_trustar_data_to_intsights_copy.py:387]: Please pay attention - you do not use the return values from `sendData` function, so nothing will be returned to the user after running this command.

File: Packs/TruSTAR/Integrations/dw_send_trustar_data_to_intsights_copy/dw_send_trustar_data_to_intsights_copy.py [hunk 7]
```diff
@@ -0,0 +1,406 @@
...
+    demisto.results(
+        {
+            "Type": entryTypes["note"],
+            "ContentsFormat": formats["text"],
+            "Contents": getCountOfCleanList,
+            "EntryContext":
+            {
+                "IndicatorValues.Length": getCountOfCleanList
+            }
+        })
```
Comment (abaumgarten) [inline Packs/TruSTAR/Integrations/dw_send_trustar_data_to_intsights_copy/dw_send_trustar_data_to_intsights_copy.py:402]: Please see my comment about `CommandResults` and use it here.

**Review by abaumgarten — COMMENTED** [review]
@rod-castillo , Thank you for your contribution!
Please see my comments below.
In addition, I would like you to pay attention to some more points as well:
1. You should run `demisto-sdk update-release-notes -i Packs/TruSTAR -u (major|minor|revision)` in order to update the pack’s version. This command will add a new release note file and will modify pack_metadata.json. You can see that error on “run validations” tests as part of our build tests. In addition, you may find more information as part of our documentation.
2. You should change integration’s name (and so for other files name), I suggest “TrustarToIntsights”. What do you think? 
3. You should add a publisher logo, you may follow instructions as written here: https://xsoar.pan.dev/docs/integrations/packs-format#author_imagepng
4. Please run demise-sdk format before pushing all of your fixes.
5. An optional one - in your python file, use a main function (will make your code “cleaner”). 
Please feel free to send me a slack message if you need help or if one of my comments not clear to you. Please let me know when you are close to finish all of those fixes and we will schedule a demo session.

File: Packs/TruSTAR/Integrations/dw_send_trustar_data_to_intsights_copy/dw_send_trustar_data_to_intsights_copy.py (hunk 2 above)
Comment (rod-castillo) [reply Packs/TruSTAR/Integrations/dw_send_trustar_data_to_intsights_copy/dw_send_trustar_data_to_intsights_copy.py:372]: Hey, thanks for giving me the documentation! I will update the code to work with the best practices over the next few days. I will commit it when it is ready!

File: Packs/TruSTAR/Integrations/dw_send_trustar_data_to_intsights_copy/dw_send_trustar_data_to_intsights_copy.yml [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+script:
+  script: ''
+  type: python
+  commands:
+  - name: trustar-get-IOC-data
+    arguments: []
+  - name: trustar-clean-IOC-data
+    arguments: []
+  - name: trustar-send-IOC-data-to-intsights
+    arguments: []
```
[the remark this answers is not in the document]
Comment (rod-castillo) [reply Packs/TruSTAR/Integrations/dw_send_trustar_data_to_intsights_copy/dw_send_trustar_data_to_intsights_copy.yml:63]: Hey! Saw your comments on the Python script, I am going to fix that in the next few days

**Comment by abaumgarten** [comment]
@rod-castillo  hi there! I would like to schedule a short call in advance, when do you estimate that you will be done with all the suggested changes? thanks

**Comment by rod-castillo** [comment]
After reevaluating how the integration is designed, I have realized that the functionality this code is trying to achieve would work much better using the main Trustar integration to get the data and a custom automation to prepare and send the Trustar data to Intsights. I will close this pull request and contribute the automation that I create when it is complete.

**Pull request closed without merging**